pydicom/filereader.py

```diff
@@ -282,6 +294,12 @@ def data_element_generator(fp,
         if length != 0xFFFFFFFF:
             # don't defer loading of Specific Character Set value as it is
             # needed immediately to get the character encoding for other tags
+            if tag_set is not None and tag not in tag_set and tag != (
```

Comment:
> AFAIK, undefined length happens only for sequence tags and pixel data

I believe that is correct.  

I'm having trouble finding the line you mention - but in any case it sounds like it is just unnecessary, _if_ we assume the file creator followed the dicom standard.  So if the line is just a harmless extra, but would help if someone did a non-compliant undefined length for some other tag, I would suggest leaving it in.